src/main/java/uk/gov/hmcts/reform/idam/health/vault/TokenResponseHandler.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.idam.health.vault;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class TokenResponseHandler implements ResponseHandler<String> {
+
+    private final static String ACCESS_TOKEN_KEY = "access_token";
+
+    private static volatile TokenResponseHandler instance;
+
+    private TokenResponseHandler() {
+    }
+
+    public static TokenResponseHandler getInstance() {
+        if (instance == null) {
+            synchronized (TokenResponseHandler.class) {
+                if (instance == null) {
+                    instance = new TokenResponseHandler();
+                }
+            }
+        }
+
+        return instance;
+    }
+
+    @Override
+    public String handleResponse(HttpResponse response) throws IOException {
+        StatusLine statusLine = response.getStatusLine();
+        if (statusLine.getStatusCode() != HttpStatus.SC_OK) {
+            throw new HttpResponseException(statusLine.getStatusCode(), statusLine.getReasonPhrase());
+        }
+
+        HttpEntity entity = response.getEntity();
+        String json = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Can we make `mapper` a `private static final` field? It can be quite expensive to initialise on every request.

Reply:
Done.